README

                     Ampoliros Application Server

                       http://www.ampoliros.com



                         AMPOLIROS INFORMATION
                         =====================



General information
-------------------

Ampoliros is an advanced and easy to use distributed PHP Web
applications platform, featuring a powerful Xml-Rpc and Soap interface.

It is suitable as an Internet/Intranet development and deployment
system. It has a very strong modular architecture and allows very fast
deployment of web solutions.


Features
--------

- Web interface. The first standard Ampoliros interface is the web
administration system.

- Xml interface for Web Services. The second Ampoliros standard interface
is the remote procedures one, through XML. The platform provides
interface for XmlRpc and Soap calls.

- Centralized sites administration. Creation, editing and remotion of the
sites can be done through a single interface.

- Centralized applications administration. Installation, update and
remotion of the applications (called "modules" in Ampoliros
terminology) can be done through a single interface.

- Extensibility of platform functions. Platform functions can be extended
through external modules and hooks.

- Extreme modularity. The whole platform is designed with modularity in
mind.

- Interaction between the applications. Every application can interact
with the other applications, through API calls, hooks, remote
procedures. The platform also provides dependencies support between
applications.

- Immediate installation and update of applications. To install an
application in the platform you only need to upload the application
file through a modules administration page. The same applies for the
update of already installed applications.

- Install applications once - use many times. When installing an
application, it can be enabled to all platform sites without
reinstalling it.

- Easy installation and immediate update of the platform. Since Ampoliros
itself is a module, it has all of modules properties and can be updated
like any other module with a single step.

- Separation of code and presentation. Application interface is
programmed with a dedicated library of functions; no HTML in code.

- Use of open standards and technologies. Ampoliros follow open standards
like SQL, XML, XML-RPC, Soap and no proprietary formats are introduced.

- Open licensing. Ampoliros is licensed with the Gnu Public License
(GPL).

- Localization support. Ampoliros supports country and language
localization, both on platform and applications levels.

- Context sensitive help. The system provides online help.

- Database abstraction. The platform provides an extensible database
abstraction layer. Current available interfaces: PostgreSQL and MySQL.

- Written in PHP language. Ampoliros is written in PHP, a language born
and specifically designed for the web.

- Cross platform. Being written in PHP language, Ampoliros can be
installed in every operating system where PHP has been ported.


License
-------

Ampoliros is released under the GPL license. See the file named LICENSE.


Requirements
------------

Serverside:
    A web server with PHP 4.3 support and a SQL server supported by Ampoliros.
    Read INSTALL file for more information.

Clientside:
    A web browser with HTML 4.0 and CSS support.

See the file named INSTALL for more details.


Installation
------------

See the file named INSTALL.


Additional Information
----------------------

Ampoliros official home page:
    http://www.ampoliros.com/

Ampoliros latest distribution:
    http://www.ampoliros.com/en/dl/download.php
    http://sourceforge.net/project/showfiles.php?group_id=21115

Ampoliros IRC channel:
    #ampoliros on irc.openprojects.net

Ampoliros mailing lists:
    none at the moment

Solarix official home page:
    http://www.solarix.it/

Ampoliros page at Solarix site:
    http://www.solarix.it/prodotti/ampoliros/

To submit a bug report:
    send an email to info@ampoliros.com

To submit a support request:
    send an email to info@ampoliros.com


Disclaimer
----------

There is no warranty, expressed or implied, associated with this product.
Use at your own risk.


CHANGES

                     Ampoliros Application Server

                       http://www.ampoliros.com



                                CHANGES
                                =======



4000 4.1 - 2004/10/15 - "Hui MVC"
---------------------
SECURITY:
FIXES:
CHANGES:
NEW FEATURES:
DEPRECATED:



4000 4.0 - 2004/07/15 - "Major refactoring"
---------------------
NEW FEATURES:
- Added new "amp4000" theme. (wuh)
- Added new "big" hui icons category. (wuh)
- Added 494 new icons in "big" category. (wuh)
- Added new com.solarix.ampoliros.core.Ampoliros class. (wuh)
- Added unit tests under com.solarix.ampoliros.test. (wuh)
- Added huixml_cdata() (com.solarix.ampoliros.hui.widgets.HuiXml) in
  place of urlencode() method. (wuh)

CHANGES:
- Changed Ampoliros generation to 4000. (wuh)
- Now Ampoliros is based on the Carthag platform. (wuh)
- Updated Crystal icons to SVG release. Added 584 more new icons. (wuh)
- Moved Semaphore class (semaphores.library) to
  com.solarix.ampoliros.process.Semaphore. (wuh)
- Moved Clipboard class (clipboard.library) to
  com.solarix.ampoliros.datatransfer.Clipboard. (wuh)
- Moved CachedItem class (cache.library) to 
  com.solarix.ampoliros.datatransfer.cache.CachedItem. (wuh)
- Moved CacheGarbageCollector (cache.library) to
  com.solarix.ampoliros.datatransfer.cache.CacheGarbageCollector. (wuh)
- Moved hui.library classes to com.solarix.ampoliros.hui.*. (wuh)
- Moved huithemes.library classes to com.solarix.ampoliros.hui.theme.*. (wuh)
- All code is now under a declare() statement. (wuh)
- Added pcntl and posix as suggested extensions. (wuh)
- Changed "Ampoliros Web Applications Platform" title to "Ampoliros
  Application Server". (wuh)
- Moved maintenance.library classes to com.solarix.ampoliros.maintenance.*.
  (wuh)
- Moved Hook class (hooks.library) to com.solarix.ampoliros.util.Hooks. (wuh)
- Renamed AmpolirosSecurityHandler() class to SecurityLayer() and moved it
  to com.solarix.ampoliros.security.SecurityLayer. (wuh)
- Moved archives.library classes to Carthag platform. (wuh)
- Moved debug.library classes to com.solarix.ampoliros.debug.* and renamed
  AmpDebugger() class to Debugger(). (wuh)
- Moved xmlrpc.library classes to com.solarix.ampoliros.webservices.xmlrpc.*.
  (wuh)
- Removed unused system.library. (wuh)
- Moved sites.library classes to com.solarix.ampoliros.site.*. (wuh)
- Moved users.library classes to com.solarix.ampoliros.site.user.*. (wuh)
- Moved logger.library classes to com.solarix.ampoliros.io.log.*. (wuh)
- Moved dblayer libraries classes to com.solarix.ampoliros.db.*. (wuh)
- Moved locale.library classes to com.solarix.ampoliros.locale.*. (wuh)
- Moved old unused stuff to ampcompatibility package. (wuh)
- Now at least PHP 5 is required. (wuh)
- Updated copyright notes date. (wuh)
- Moved XMLParser class (xml.library) to com.solarix.ampoliros.xml.XMLParser.
  (wuh)
- Applied minor changes for PHP 5 compatibility. (wuh)
- Moved modules.library and modulesbase.library classes to
  com.solarix.ampoliros.module.*. (wuh)
- Strongly refactored maintaining compatibility. (wuh)
- Changed handling of /ampcgi/ alias during setup. (wuh)
- Renamed ModuleElementTypes class to
  com.solarix.ampoliros.module.ModuleElementFactory. (wuh)
- Moved soap.library to ampcompatibility. (wuh)
- Updated some stuff in extras directory. (wuh)
- Changed "module" term in "application". (wuh)
- Renamed root/modules.php to root/applications.php. (wuh)
- Moved HelpNode class to com.solarix.ampoliros.help.HelpNode. (wuh)
- Moved AmpConfig class to com.solarix.ampoliros.core.AmpConfig. (wuh)
- Lot of optimizations in Hui and Hui widgets. (wuh)
- Hui validators are now called once even if there are more dispatchers. (wuh)
- Moved root initialization code to Ampoliros::startRoot(). (wuh)
- Moved site initialization code to Ampoliros::startSite(). (wuh)
- Moved maintenance run script code to Ampoliros::startMaintenance(). (wuh)
- Moved ampconfig.php code to ampoliros.php. (wuh)
- Removed amp3000 Hui theme. (wuh)
- Removed CACHE_PATH define. (wuh)

FIXES:
- Fixed "frame" attribute in toolbar widget when set to false. (wuh)
- Fixed wrong size limit in username length in users table. (wuh)
- Unlock script (/ampcgi/unlock.php) is no more recorded as last root
  page opened. (wuh)
- Fixed wrong parameters to Permissions class in adminpage.handler and
  admingroup.handler. (wuh)

SECURITY:

DEPRECATED:
- hui.library. (wuh)
- huithemes.library. (wuh)
- semaphores.library. (wuh)
- clipboard.library. (wuh)
- cacheditem.library. (wuh)
- maintenance.library. (wuh)
- hooks.library. (wuh)
- security.library. (wuh)
- AmpolirosSecurityHandler() class. (wuh)
- archives.library. (wuh)
- debug.library. (wuh)
- AmpDebugger() class. (wuh)
- xmlrpc.library. (wuh)
- users.library. (wuh)
- sites.library. (wuh)
- logger.library. (wuh)
- dblayer.library. (wuh)
- dblayer_parser.library. (wuh)
- dblayer_*.layer. (wuh)
- locale.library. (wuh)
- xml.library. (wuh)
- modules.library. (wuh)
- modulesbase.library. (wuh)
- help.library. (wuh)
- ampoliros.library. (wuh)
- amproot.php. (wuh)
- ampsite.php. (wuh)
- ampconfig.php. (wuh)
- LOG_PATH define. (wuh)
- CATALOG_PATH define. (wuh)
- BIN_PATH define. (wuh)
- HELP_PATH define. (wuh)
- HTTPD_GROUP and HTTPD_USER defines. (wuh)



3000 3.7 - 2004/06/15 - "PHP 5 / Ampoliros 4000 bridge"
---------------------
NEW FEATURES:
- Added multiple rows feature in tab widget. (wuh)
- Added support for "big" icons category. (wuh)
- Now Ampoliros 3000 is PHP 5 compliant. (wuh)

FIXES:
- Fixed bad hint functionality in some Hui widgets. (wuh)
- Fixed handling of AM/PM in DbLayer::GetDateArrayFromTimestamp(). (wuh)
- Fixed LocaleCountry() logged errors during setup phase. (wuh)
- Fixed minor issues with PHP5 changes. (wuh)



3000 3.6 - 2003/12/15 - "Locale reorganization"
---------------------
NEW FEATURES:
- Added "About Ampoliros" menu. (wuh)
- Added "TreeVMenu" Hui widget. (wuh)
- Added RPM spec file and build system. (wuh)

www/sum.php
<?php
/*
 *
 *                    Ampoliros Application Server
 *
 *                      http://www.ampoliros.com
 *
 *
 *
 *   Copyright (C) 2000-2004 Solarix
 *
 *   This program is free software; you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation; either version 2 of the License, or
 *   (at your option) any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program; if not, write to the Free Software
 *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *
 */
// $Id: sum.php,v 1.20 2004-07-08 15:04:26 alex Exp $

require ('./auth.php');

import('com.solarix.ampoliros.locale.Locale');
import('com.solarix.ampoliros.hui.Hui');
import('com.solarix.ampoliros.hui.HuiDispatcher');

header('P3P: CP="CUR ADM OUR NOR STA NID"');
header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
header('Last-Modified: '.gmdate('D, d M Y H:i:s'));
header('Cache-control: no-cache, must-revalidate');
header('Pragma: no-cache');

function sum_page() {
    global $gEnv;

    $amp_locale = new Locale('amp_root_menu', AMP_LANG);
    import('com.solarix.ampoliros.module.ModuleConfig');
    $mod_cfg = new ModuleConfig($gEnv['root']['db'], 'ampoliros');

    $hui = new Hui($gEnv['root']['db'], TRUE);
    $hui -> LoadWidget('table');
    $hui -> LoadWidget('page');
    $hui -> LoadWidget('vertgroup');
    $hui -> LoadWidget('vertframe');
    $hui -> LoadWidget('treemenu');

    $hui_page = new HuiPage('page', array('title' => 'Ampoliros'. (strlen(AMP_HOST) ? ' - '.AMP_HOST. (strlen(AMP_DOMAIN) ? '.'.AMP_DOMAIN : '') : ''), 'border' => 'false'));
    $hui_page -> mArgs['background'] = $hui_page -> mThemeHandler -> mStyle['menuback'];
    $hui_mainvertgroup = new HuiVertGroup('mainvertgroup');

    $el[1]['groupname'] = 'Ampoliros';

    $cont = 1;

    $query = & $gEnv['root']['db'] -> Execute('SELECT id FROM sites');
    if ($query -> NumRows()) {
        $el[1]['groupelements'][$cont]['name'] = $amp_locale -> GetStr('siteadmin');
        $el[1]['groupelements'][$cont]['image'] = $hui_page -> mThemeHandler -> mStyle['siteaccess'];
        $el[1]['groupelements'][$cont]['action'] = 'admin/';
        $el[1]['groupelements'][$cont]['themesized'] = 'true';

        $cont ++;
    }

    $el[1]['groupelements'][$cont]['name'] = $amp_locale -> GetStr('rootadmin');
    $el[1]['groupelements'][$cont]['image'] = $hui_page -> mThemeHandler -> mStyle['rootaccess'];
    $el[1]['groupelements'][$cont]['action'] = 'root/';
    $el[1]['groupelements'][$cont]['themesized'] = 'true';

    if ($mod_cfg -> GetKey('ampoliros-link-disabled') != '1') {
        $el[1]['groupelements'][++ $cont]['name'] = $amp_locale -> GetStr('amphome');
        $el[1]['groupelements'][$cont]['image'] = $hui_page -> mThemeHandler -> mStyle['ampminilogo'];
        $el[1]['groupelements'][$cont]['action'] = 'http://www.ampoliros.com/';
        $el[1]['groupelements'][$cont]['target'] = 'op';
        $el[1]['groupelements'][$cont]['themesized'] = 'true';
    }

    if ($mod_cfg -> GetKey('solarix-link-disabled') != '1') {
        $el[1]['groupelements'][++ $cont]['name'] = $amp_locale -> GetStr('solarixhome');
        $el[1]['groupelements'][$cont]['image'] = $hui_page -> mThemeHandler -> mStyle['solarixminilogo'];
        $el[1]['groupelements'][$cont]['action'] = 'http://www.solarix.biz/';
        $el[1]['groupelements'][$cont]['target'] = 'op';
        $el[1]['groupelements'][$cont]['themesized'] = 'true';
    }

    if ($mod_cfg -> GetKey('oem-link-disabled') != '1') {
        $oem_link_filename = $mod_cfg -> GetKey('oem-link-filename');

        if (strlen($oem_link_filename) and file_exists(CGI_PATH.$oem_link_filename)) {
            $el[1]['groupelements'][++ $cont]['name'] = $mod_cfg -> GetKey('oem-name');
            $el[1]['groupelements'][$cont]['image'] = CGI_URL.$oem_link_filename;
            $el[1]['groupelements'][$cont]['action'] = $mod_cfg -> GetKey('oem-url');
            $el[1]['groupelements'][$cont]['target'] = 'parent';
            $el[1]['groupelements'][$cont]['themesized'] = 'false';
        }
    }

    $hui_vertframe = new HuiVertFrame('vertframe');
    $hui_vertframe -> AddChild(new HuiTreeMenu('treemenu', array('elements' => $el, 'width' => '120', 'target' => 'parent', 'allgroupsactive' => 'true')));

    $hui_mainvertgroup -> AddChild($hui_vertframe);
    $hui_page -> AddChild($hui_mainvertgroup);
    $hui -> AddChild($hui_page);
    $hui -> Render();
}

function setup_entry($phaseMark, $phaseCompleted, $phaseName, & $hui_table, $row) {
    global $hui_page, $progress, $phases;

    if (file_exists(TMP_PATH.$phaseMark)) {
        $ball_icon = $hui_page -> mThemeHandler -> mStyle['goldball'];
        $font_color = 'yellow';
        $pre = '<b>';
        $post = '</b>';
    } else
        if (!file_exists(TMP_PATH.$phaseCompleted)) {
            $ball_icon = $hui_page -> mThemeHandler -> mStyle['redball'];
            $font_color = 'black';
            $pre = '';
            $post = '';
        } else {
            $ball_icon = $hui_page -> mThemeHandler -> mStyle['greenball'];
            $font_color = 'black';
            $pre = '';
            $post = '';
            $progress = $row +1;
        }

    $hui_table -> AddChild(new HuiImage('statusimage'.$row, array('imageurl' => $ball_icon)), $row, 0);
    $hui_table -> AddChild(new HuiLabel('phaselabel'.$row, array('label' => $pre.$phaseName.$post, 'nowrap' => 'false')), $row, 1);
}

// Checks if Ampoliros is in setup phase
//
if (!defined('AMPOLIROS_SETUP_PHASE')) {
    sum_page();
} else {
    global $gEnv;

    $progress = 0;

    $amp_locale = new Locale('amp_misc_install', AMP_LANG);

    $hui = new Hui($gEnv['root']['db'], TRUE);
    $hui -> LoadWidget('image');
    $hui -> LoadWidget('label');
    $hui -> LoadWidget('table');
    $hui -> LoadWidget('page');
    $hui -> LoadWidget('vertgroup');
    $hui -> LoadWidget('vertframe');
    $hui -> LoadWidget('treemenu');
    $hui -> LoadWidget('progressbar');

    $hui_page = new HuiPage('page', array('title' => 'Ampoliros'. (strlen(AMP_HOST) ? ' - '.AMP_HOST. (strlen(AMP_DOMAIN) ? '.'.AMP_DOMAIN : '') : '')));
    $hui_page -> mArgs['background'] = $hui_page -> mThemeHandler -> mStyle['menuback'];
    $hui_mainvertgroup = new HuiVertGroup('mainvertgroup');

    $headers[1]['label'] = $amp_locale -> GetStr('setupphase_header');

    $hui_table = new HuiTable('sumtable', array('headers' => $headers));

    $phase = 0;
    $phases = 13;

    setup_entry('.checkingsystem', '.systemchecked', $amp_locale -> GetStr('systemcheckphase_label'), $hui_table, $phase ++);
    setup_entry('.installingfiles', '.filesinstalled', $amp_locale -> GetStr('filesinstallphase_label'), $hui_table, $phase ++);
    setup_entry('.settingedition', '.editionset', $amp_locale -> GetStr('editionchoicephase_label'), $hui_table, $phase ++);
    setup_entry('.creatingdblayers', '.dblayerscreated', $amp_locale -> GetStr('dblayersphase_label'), $hui_table, $phase ++);
    setup_entry('.creatingdb', '.dbcreated', $amp_locale -> GetStr('amprootdbphase_label'), $hui_table, $phase ++);
    setup_entry('.initializingcomponents', '.componentsinitialized', $amp_locale -> GetStr('ampcomponentsphase_label'), $hui_table, $phase ++);
    setup_entry('.settingamphost', '.amphostset', $amp_locale -> GetStr('amphostchoicephase_label'), $hui_table, $phase ++);
    setup_entry('.settingcountry', '.countryset', $amp_locale -> GetStr('countrychoicephase_label'), $hui_table, $phase ++);
    setup_entry('.settinglanguage', '.languageset', $amp_locale -> GetStr('languagechoicephase_label'), $hui_table, $phase ++);
    setup_entry('.settingpassword', '.passwordset', $amp_locale -> GetStr('passwordphase_label'), $hui_table, $phase ++);
    setup_entry('.settingampcentral', '.ampcentralset', $amp_locale -> GetStr('ampcentralphase_label'), $hui_table, $phase ++);
    setup_entry('.cleaningup', '.cleanedup', $amp_locale -> GetStr('cleanupphase_label'), $hui_table, $phase ++);
    setup_entry('.finishingsetup', '.setupfinished', $amp_locale -> GetStr('finishphase_label'), $hui_table, $phase ++);

    $hui_mainvertgroup -> AddChild($hui_table);
    $hui_mainvertgroup -> AddChild(new HuiProgressBar('progress', array('progress' => $progress, 'totalsteps' => $phases)));
    $hui_page -> AddChild($hui_mainvertgroup);
    $hui -> AddChild($hui_page);
    $hui -> Render();
}
?>
